BanchoSharp/Multiplayer/MultiplayerLobby.cs

```diff
@@ -408,6 +372,72 @@ private void UpdateLobbyFromBanchoBotSettingsResponse(string banchoBotResponse)
 
 			OnPlayerSlotMove?.Invoke(new PlayerSlotMoveEventArgs(player, previousSlot, slotNum));
 		}
+		else if (banchoBotResponse.StartsWith("Players: "))
+		{
+			var playerCount = int.Parse(banchoBotResponse.Split("Players: ")[1]);
+
+			if (playerCount == 0)
+			{
+				// The "Players: <count>" should only come after a !mp settings request, so if we've gotten this, 
+				// and the count is 0, the "!mp settings" should have finished.
+
+				OnSettingsUpdated?.Invoke();
+			}
+			else
+			{
+				_playersRemainingCount = playerCount;
+			}
+		}
+		else if (banchoBotResponse.StartsWith("Slot "))
+		{
+			var slot = int.Parse((banchoBotResponse[..8].Where(c => char.IsDigit(c)).ToArray()));
+			var state = banchoBotResponse.Substring(8, 10).TrimEnd();
+
+			// Find the first ' ' after the URL, since the URL is not padded with any spaces.
+			var playerNameBegin = banchoBotResponse.IndexOf(' ', banchoBotResponse.IndexOf("/u/")) + 1;
+
+			var url = banchoBotResponse[18..(playerNameBegin - 1)];
+			var name = banchoBotResponse.Substring(playerNameBegin, 16).TrimEnd();
+			var info = banchoBotResponse.Length > (playerNameBegin + 16) ? banchoBotResponse[(playerNameBegin + 16)..] : null;
+
+			var player = FindPlayer(name);
+
+			if (player is null)
+			{
+				OnPlayerJoined?.Invoke(new MultiplayerPlayer(name, slot, TeamColor.None));
+			}
+			else
+			{
+				if (player.Slot != slot)
+				{
+					var previousSlot = player.Slot;
+
+					player.Slot = slot;
+
+					OnPlayerSlotMove?.Invoke(new PlayerSlotMoveEventArgs(player, previousSlot, slot));
+				}
+			}
+
+			if (--_playersRemainingCount == 0)
```

Comment:
Using a prefix decrement inside of an if statement is weird to me, I prefer to not use this form of arithmetic. I also can't tell why this variable is being subtracted here.